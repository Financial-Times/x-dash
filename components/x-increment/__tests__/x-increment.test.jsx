const { h } = require('@financial-times/x-engine');
const { mount } = require('@financial-times/x-test-utils/enzyme');

const { Increment } = require('../');

describe('x-increment', () => {
	it('should increment when action is triggered', async () => {
		const subject = mount(<Increment count={1} />);
		await subject.find('BaseIncrement').prop('actions').increment();

		expect(subject.find('span').text()).toEqual('2');
	});

	it('should increment by amount from action arg', async () => {
		const subject = mount(<Increment count={1} />);
		await subject.find('BaseIncrement').prop('actions').increment({ amount: 2 });

		expect(subject.find('span').text()).toEqual('3');
	});

	it('should increment when clicked, waiting for timeout', async () => {
		const subject = mount(<Increment count={1} timeout={1000} />);
		const start = Date.now();

		await subject.find('button').prop('onClick')();

		expect(Date.now() - start).toBeCloseTo(1000, -2); // negative precision â‡’ left of decimal point
		expect(subject.find('span').text()).toEqual('2');
	});
});
