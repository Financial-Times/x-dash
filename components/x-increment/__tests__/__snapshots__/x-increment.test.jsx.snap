// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`x-increment snapshots renders a default Async nodejs 1`] = `
<div>
  <span>
    1
  </span>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Increment
  </button>
</div>
`;

exports[`x-increment snapshots renders a default Increment nodejs 1`] = `
<div>
  <span>
    1
  </span>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Increment
  </button>
</div>
`;
